Dockerfile

```diff
@@ -18,161 +19,232 @@
 # SPDX-License-Identifier: Apache-2.0
 # License-Filename: LICENSE
 
-FROM adoptopenjdk/openjdk11:jdk-11.0.9_11.1-alpine-slim AS build
 
-ARG ORT_VERSION="DOCKER-SNAPSHOT"
+#------------------------------------------------------------------------
+# Build components
+FROM ubuntu:focal AS build
+
+# http://bugs.python.org/issue19846
+ENV LANG C.UTF-8
 
-# Apk install commands.
-RUN apk add --no-cache \
-        # Required for Node.js to build the reporter-web-app.
-        libstdc++ \
-        # Required to allow to download via a proxy with a self-signed certificate.
-        ca-certificates \
-        coreutils \
-        openssl
+USER root
 
+#------------------------------------------------------------------------
+# Basic intro
+RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt \
+    apt update \
+    && DEBIAN_FRONTEND=noninteractive apt upgrade -y --no-install-recommends \
+    && apt install -y --no-install-recommends bash ca-certificates curl gnupg wget
+
+#------------------------------------------------------------------------
+# Ubuntun build toolchain
+RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt \
+    apt update \
+    && DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends \
+        build-essential \
+        dirmngr \
+		dpkg-dev \
+        git \
+		libbluetooth-dev \
+		libbz2-dev \
+		libc6-dev \
+		libexpat1-dev \
+		libffi-dev \
+        libgmp-dev \
+		libgdbm-dev \
+		liblzma-dev \
+        libmpdec-dev \
+		libncursesw5-dev \
+		libreadline-dev \
+		libsqlite3-dev \
+		libssl-dev \
+		make \
+        netbase \
+        openjdk-11-jdk \
+		tk-dev \
+        tzdata \
+        unzip \
+		uuid-dev \
+		xz-utils \
+		zlib1g-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /work
+
+# Copy the necessary bash resource to have paths
+# for different languages
+COPY docker/root-bashrc /root/.bashrc
+RUN mkdir -p /root/.bash_includes
+
+#------------------------------------------------------------------------
+# Python
+# Using pyenv and make 3.8 default
+ENV CONAN_VERSION=1.38.0
+ENV SCANCODE_VERSION=21.6.7
+# Python pyenv
+ENV PYENV_ROOT=/opt/python
+RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+ENV PATH=/opt/python/bin:$PATH
+RUN pyenv install 3.8.11
```

Comment:
I propose to replace 3.8.11 in this file with a single PYTHON_VERSION environment variable so users can more easily update it

Reply:
Done as an ARG